jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt

```diff
@@ -15,35 +15,33 @@ import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VfsUtil
 import com.intellij.openapi.vfs.VirtualFile
 import com.intellij.openapi.vfs.VirtualFileWrapper
-import com.intellij.ui.AnActionButton
 import com.intellij.ui.treeStructure.treetable.TreeTable
 import org.jetbrains.annotations.TestOnly
 import software.amazon.awssdk.core.sync.ResponseTransformer
 import software.amazon.awssdk.services.s3.S3Client
 import software.amazon.awssdk.services.s3.model.GetObjectRequest
-import software.aws.toolkits.core.telemetry.TelemetryNamespace
+import software.aws.toolkits.jetbrains.components.telemetry.ActionButtonWrapper
 import software.aws.toolkits.jetbrains.core.AwsClientManager
 import software.aws.toolkits.jetbrains.services.s3.S3VirtualBucket
 import software.aws.toolkits.jetbrains.services.s3.bucketEditor.S3KeyNode
-import software.aws.toolkits.jetbrains.services.telemetry.TelemetryService
 import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.resources.message
 import javax.swing.tree.DefaultMutableTreeNode
 
 class DownloadObjectAction(
     private val treeTable: TreeTable,
     val bucket: S3VirtualBucket,
     private val fileChooser: FileChooserFactory
-) : AnActionButton("Download Object", null, AllIcons.Actions.Download), TelemetryNamespace {
+) : ActionButtonWrapper(message("s3.download.object.action"), null, AllIcons.Actions.Download) {
 
     constructor(treeTable: TreeTable, bucket: S3VirtualBucket) : this(treeTable, bucket, FileChooserFactory.getInstance())
 
     @Suppress("unused")
-
-    override fun actionPerformed(e: AnActionEvent) {
+    override fun doActionPerformed(e: AnActionEvent) {
         val project = e.getRequiredData(LangDataKeys.PROJECT)
         val client: S3Client = AwsClientManager.getInstance(project).getClient()
```

Comment:
Unlike the explorer node, the S3 client here might not be created correctly. As you open the bucket editor, you need to retain the credentials and region for that bucket into the editor since the active credentials and region could be changed outside. This client should be built from the retained credentials and region instead.